bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/things/VeluxExistingProducts.java

```diff
@@ -123,8 +123,8 @@ public boolean update(ProductBridgeIndex bridgeProductIndex, int productState, i
             return false;
         }
         VeluxProduct thisProduct = this.get(bridgeProductIndex);
-        if (thisProduct.setState(productState) || thisProduct.setCurrentPosition(productPosition)
-                || thisProduct.setTarget(productTarget)) {
+        if (thisProduct.setState(productState) | thisProduct.setCurrentPosition(productPosition)
+                | thisProduct.setTarget(productTarget)) {
```

Comment:
If I see correctly, this doesn't change any semantic. I think it's logical to use logical ORs here. What's the purpose of this change?

Reply:
@fwolter IMHO this change is essential. In a Java `if` statement based on `||` the compiler does logical short circuit optimisations; this means that if the leftmost function evaluates to `true`, then any functions further right need not be, **_and therefore are not_**, evaluated. OTOH replacing the `||` operator by `|` prevents logical short circuit optimisations, and so it forces all three property setters to be called.

Reply:
Ok, I see. Can you add a comment that the next contributor/reviewer doesn't stumble upon it?

Reply:
Ok.